pandas/core/generic.py

```diff
@@ -10351,15 +10351,22 @@ def _logical_func(
                 name, func, axis=0, bool_only=bool_only, skipna=skipna, **kwargs
             )
             return res._logical_func(name, func, skipna=skipna, **kwargs)
-
-        return self._reduce(
-            func,
-            name=name,
-            axis=axis,
-            skipna=skipna,
-            numeric_only=bool_only,
-            filter_type="bool",
-        )
+        try:
+            return self._reduce(
+                func,
+                name=name,
+                axis=axis,
+                skipna=skipna,
+                numeric_only=bool_only,
+                filter_type="bool",
+            )
+        except NotImplementedError as exc:
```

Comment:
this is not a good soln as making this even more complicated (_reduce itself is quite complicated).

Reply:
what actually needs to happen is we pass bool_only thru, but this increases the scope of this PR greatly.

let's just match the current message with an xfail and open an issue.

Reply:
Sorry @jreback - not sure I understand.

I will revert the last commit, but I think we only need to either 1) match the current message or 2) put that the message should be bool_only and mark with xfail

Reply:
you can match the message for now with a TODO that its wrong. pls open an issue about this. The patch that you added is not correct, its just band-aiding a very complex routine; its not in the correct place, fixing is out of scope.

Reply:
Ok, #38810 for fixing the error message. Removed the bad fix and asserts the old error message and added a TODO